Add .i extension for preprocessed C files to known file extensions

cschlote opened the pull request:
Added .i as extension for preprocessed C files, which can be fed directly
to DMD. The file is then processed as C source file.

Supporting .i as file exzension allows to preprocess C code with GCC -E and
store the output to some file for further processing by DMD.

cschlote committed:
Add .i extension for preprocessed C files to known set of file extensions

Added .i as extension for preprocessed C files, which can be fed directly
to DMD. The file is then processed as C source file.

Supporting .i as file exzension allows to preprocess C code with GCC -E and
store the output to some file for further processing by DMD.

thewilsonator commented on src/dmd/dmodule.d:
```diff
@@ -1042,7 +1047,7 @@ extern (C++) final class Module : Package
 
         /* If it has the extension ".c", it is a "C" file.
          */
-        if (FileName.equalsExt(arg, c_ext))
+        if (FileName.equalsExt(arg, c_ext) || FileName.equalsExt(arg, i_ext))
```
Please update the comment too.

cschlote replied:
Comment updated,
Adding a test is WIP. I'm working through the test infrastructure of DMD and try to find the proper place for it. Is there any 'importc' releated test, yet?

thewilsonator reviewed (commented):
Please also add a test case

WalterBright commented:
Needs a changelog entry, too.

cschlote committed:
Update existing Changelog on 'importc' feature.

Add some example on how to use the '.i' suffix instead of the '.c' suffix
for C files using C includes. Result of C preprocessor can be saved as
'.i' file and used as input to DMD.

cschlote committed:
Add some first simple test for importc with .i files passed on cmdline

cschlote committed:
Hide preprocessed C files

cschlote commented:
Testcase added: ./run.d runnable/importc-test1.sh

Commit:
Changed importc test to use .i.in file to fake preprocessor

TODO: Find some preprocessor, which works on Windows and Max targets.

cschlote commented:
There is an open issue with the C compiler to use on Windows and Mac. Faked input '.i' file as workaround.
